### PR TITLE
Manage: Identify and remove all instances of "Automatic Rejections"

### PR DESCRIPTION
**Pull request by vassyz**
This commit addresses a few scenarios where we were displaying information related to reject by default date.
It also adds a concern to the `ApplicationChoice` model to help with the date calculations.

### Content below purple ‘Received' status change to: Received [36] days ago
Screenshot 2023-10-06 at 17 32 02

### Content below green ‘Offered’ status change to: Offer made [53] days ago
Screenshot 2023-10-06 at 17 32 29

### Content below yellow ‘Interviewing’ status change to: Received [36] days ago
Screenshot 2023-10-06 at 17 32 48

### Set up an interview or make a decision inset text within an application, change to: This application was received 80 days ago. You should try and respond to the candidate within 30 days.
Screenshot 2023-10-06 at 17 33 32

### After an offer has been made inset text within an application, change to: You made this offer [10] days ago. Most candidates respond to offers within 15 working days. The candidate will receive reminders to respond.
Screenshot 2023-10-06 at 17 33 57

### Remove inset text of Date field and change to: For example, 30 11 2023
Screenshot 2023-10-06 at 17 36 14

## Guidance to review
They way I'm handling the `days_since` methods might not be to everyone's taste. Happy to change based on feedback.

## Link to Trello card
https://trello.com/c/vaRJnjiz/714-manage-identify-and-remove-all-instances-of-automatic-rejections

### CODE REVIEW

**Comment by vassyz** [comment]
[Manage: Identify and remove all instances of "Automatic Rejections"](https://trello.com/c/vaRJnjiz/714-manage-identify-and-remove-all-instances-of-automatic-rejections)

File: app/components/provider_interface/application_header_components/offer_will_be_declined_by_default_component.html.erb [hunk 1]
```diff
@@ -2,9 +2,5 @@
   Waiting for candidate’s response
 </h2>
 <p class="govuk-body">
-  <% if application_choice.continuous_applications? %>
-    <%= continuous_applications_offer_text %>
-  <% else %>
```
Comment (JR-G) [inline app/components/provider_interface/application_header_components/offer_will_be_declined_by_default_component.html.erb:7]: We still have some candidates with offers for 2023 – they have till ~12th oct until DBD

File: app/models/concerns/dateable.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# When included in a model, it will create a method for each date field passed in that returns the number of days since
+# the date field was set.
+
+# For example, if you have a model with a field called `submitted_at`, you can call `days_since_submitted` on an
+# instance of that model
+
+# If you want to add a date field to a model, you can do so like this:
+# class MyModel < ApplicationRecord
+#   dateable :submitted_at, :updated_at
+# end
```
Comment (JR-G) [inline app/models/concerns/dateable.rb:10]: This is looking good! Thoughts on putting this text into a markdown doc ?

**Review by MaeveRoseveare — COMMENTED** [review]
I noticed that when I logged in as a provider user without the permissions to make decisions, I saw this text on a received application, still mentioning reject by default:

Screenshot 2023-10-09 at 09 27 49

When I changed the permissions to being able to make decisions, the text changed to the new one where we do not mention RBD.

Screenshot 2023-10-09 at 09 28 45

This is the same for 'Interviewing' status:

Screenshot 2023-10-09 at 09 32 23

File: app/models/concerns/dateable.rb (hunk 2 above)
Comment (vassyz) [reply app/models/concerns/dateable.rb:10]: To be addressed at a later date.

File: app/components/provider_interface/application_header_components/offer_will_be_declined_by_default_component.html.erb (hunk 1 above)
Comment (vassyz) [reply app/components/provider_interface/application_header_components/offer_will_be_declined_by_default_component.html.erb]: Done in https://github.com/DFE-Digital/apply-for-teacher-training/pull/8644/commits/f33140e68ab3580ac2df317d4ece637e48893e13. Added it as a separate commit as we can re-revert it later.

**Review by JR-G — APPROVED** [review]

**Pull request merged**